pkg/systemd/reporting.jsx

```diff
@@ -307,184 +319,124 @@ class BusWorkflowRow extends React.Component {
-    }
-
-    _onReportButtonClick(event) {
-        this.setState({ problemState: ProblemState.UNREPORTABLE });
+    };
 
-        this.setState({
-            message: _("Waiting to start…"),
-            problemState: ProblemState.REPORTING,
-        });
+    const onReportButtonClick = (_event) => {
+        setMessage(_("Waiting to start…"));
+        setProblemState(ProblemState.REPORTING);
 
-        this.props.client
-                .wait()
+        client.wait()
                 .catch(exception => console.error(cockpit.format("Channel for reportd D-Bus client closed: $0", exception.problem || exception.message)))
-                .then(() => this._createTask(this.props.client))
+                .then(() => createTask(client))
                 .catch(exception => {
-                    const message = cockpit.format("reportd task could not be created: $0", (exception.problem || exception.message));
-
-                    this.setState({
-                        message,
-                        problemState: ProblemState.REPORTABLE,
-                    });
-                    console.error(message);
-                });
-    }
-
-    render() {
-        return <WorkflowRow label={this.state.label}
-                            message={this.state.message}
-                            onCancelButtonClick={this._onCancelButtonClick}
-                            onReportButtonClick={this._onReportButtonClick}
-                            problemState={this.state.problemState}
-                            reportLinks={this.state.reportLinks}
-        />;
-    }
-
-    updateStatusFromBus() {
-        const on_get_properties = properties => {
-            if (!properties[0].CanBeReported.v) {
-                this.setState({ problemState: ProblemState.UNREPORTABLE });
-
-                return;
-            }
-
-            const reportLinks = [];
-            let reported = false;
-
-            for (const report of properties[0].Reports.v) {
-                if (!("WORKFLOW" in report[1])) {
-                    continue;
-                }
-                if (this.props.workflow[0] !== report[1].WORKFLOW.v.v) {
-                    continue;
-                }
-                if (report[0] === "ABRT Server" || report[0] === "uReport") {
-                    continue;
-                }
-                if ("URL" in report[1]) {
-                    reportLinks.push(report[1].URL.v.v);
-                }
-
-                reported = true;
-            }
-
-            if (reported) {
-                this.setState({
-                    problemState: ProblemState.REPORTED,
-                    reportLinks,
+                    const newMessage = cockpit.format("reportd task could not be created: $0", (exception.problem || exception.message));
+                    setMessage(newMessage);
+                    setProblemState(ProblemState.REPORTABLE);
+                    console.error(newMessage);
```

Comment:
These 4 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19403-20231002-082424-b39d5ff6-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)